rtutorial4/rtutorial4.Rmd

```diff
@@ -0,0 +1,414 @@
+ggplot(diamonds, aes(carat,price)) + geom_point()
+```
+
+You can also plot multiple layers of data on the same plot.
+```{r}
+
+ggplot(diamonds, aes(carat,price)) + geom_point() + geom_smooth()
+
+```
+
+Let's finish the plot with some descriptive labels.
+
+```{r}
+
+ggplot(diamonds, aes(carat,price)) + geom_point() + geom_smooth() + ggtitle("Diamond prices by carat") + labs(x="Carat", y="Price (USD)")
+
+```
+
+Now you've mastered building plots in layers.
+
+### Customizing aesthetics 
+There are multiple aesthetic parameters that can be customized in ggplots. This includes: color, fill, linetype, size, shape, font, and more. It just depends on which `geom` you are working with. We will explore some of these graphical parameters further as this tutorial introduces different geoms. [Here](https://ggplot2.tidyverse.org/articles/ggplot2-specs.html) is a vignette about aesthetic customization in ggplot2.
+
+
+### Different geoms
+
+```{r eval=FALSE}
+
+geom_col()
+geom_point()
+geom_line()
+geom_smooth()
+geom_histogram()
+geom_boxplot()
+geom_text()
+geom_density()
+geom_errorbar()
+geom_hline()
+geom_abline()
+
+```
+
+## Bar plots
+
+Bar plots are great for showing frequencies or proportions across different groups. Let's return to the otter dataset we analyzed last week. First, let's calculate the total number of otters observed per site and then plot this in a bargraph with `ggplot2`.
+```{r}
+otter <- read.csv("https://maddiebrown.github.io/ANTH630/data/sea_otter_counts_2017&2018_CLEANDATA.csv")
+notterpersite <- aggregate(x=n_otter ~ site_name, FUN=sum, data=otter)
+ggplot(notterpersite, aes(x=site_name, y=n_otter)) + geom_col()
+```
+
+### Try it
+Hmm...there are way too many columns in the plot above to make it a meaningful barchart. Subset out the top 5 sites with the most otter sightings and recreate the above chart. 
+
+*To get past this exercise, you'll need to submit the code exactly as it is in the solution with the same variable names.*
+```{r prepare-otter}
+otter <- read.csv("https://maddiebrown.github.io/ANTH630/data/sea_otter_counts_2017&2018_CLEANDATA.csv")
+notterpersite <- aggregate(x=n_otter ~ site_name, FUN=sum, data=otter)
+ggplot(notterpersite, aes(x=site_name, y=n_otter)) + geom_col()
+```
+
+```{r exercise1-1, exercise.setup="prepare-otter", exercise = TRUE}
+# notterpersite <-
+# top5 <-
+```
+
+```{r exercise1-1-solution}
+notterpersite<- notterpersite[order(notterpersite$n_otter,decreasing=T),]
+top5 <- notterpersite[1:5,]
+ggplot(top5, aes(x=site_name, y=n_otter)) + geom_col()
+```
+
+```{r exercise1-1-check}
+grade_code()
+```
+
+
+Alright, this figure is looking better. But suppose we want to reorder the categories on the X-axis such that they are in order from fewest to greatest number of otter sightings?
+
+We can reorder the cateogries according to the value in the n_otter column. First, inspect the structure of the site_name data. You'll notice that the remaining categories are in alphabetical order. 
+
```

Comment:
This section had talk about factors and droplevels(). I removed that part of the content because the top5 vector was a character vector and didn't have factor levels as far as I could see. Not sure what changed, probably in the original the factors were there. `droplevels()` was causing errors both in this version and the copy of the original tutorial that I had running since there weren't factors.